examples/recursion-limit.ipynb

```diff
@@ -0,0 +1,172 @@
+      "text/plain": [
+       "<IPython.core.display.Image object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from IPython.display import Image, display\n",
+    "\n",
+    "display(Image(graph.get_graph().draw_mermaid_png()))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we can see, our graph will execute nodes `b` and `c` in parallel (i.e. in a single super-step), which means that if we run this graph it should take exactly 3 steps. We can set the recursion limit to 3 first to check that it fails (the recursion limit is inclusive, so if the limit is 3 the graph will fail when it reaches step 3) as expected: "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Adding I'm A to []\n",
+      "Adding I'm B to [\"I'm A\"]\n",
+      "Adding I'm C to [\"I'm A\"]\n",
+      "Adding I'm D to [\"I'm A\", \"I'm B\", \"I'm C\"]\n",
+      "Recursion Error\n"
+     ]
+    }
+   ],
+   "source": [
+    "from langgraph.errors import GraphRecursionError\n",
+    "\n",
+    "try:\n",
+    "    graph.invoke({\"aggregate\": []},{\"recursion_limit\":3})\n",
+    "except GraphRecursionError:\n",
+    "    print(\"Recursion Error\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Success! The graph failed as expected - now let's test setting the recursion limit to 4 and ensure that the graph succeeds in this case:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Adding I'm A to []\n",
+      "Adding I'm B to [\"I'm A\"]\n",
+      "Adding I'm C to [\"I'm A\"]\n",
+      "Adding I'm D to [\"I'm A\", \"I'm B\", \"I'm C\"]\n"
+     ]
+    }
+   ],
+   "source": [
+    "try:\n",
+    "    graph.invoke({\"aggregate\": []},{\"recursion_limit\":4})\n",
+    "except GraphRecursionError:\n",
+    "    print(\"Recursion Error\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Perfect, just as we expected the graph runs succesfully in this case. \n",
+    "\n",
+    "Setting the correct recursion limit can be important for minimizing unneccesary API costs, and also for avoiding undesired node loops. "
```

Comment:
how about "Setting recursion limit is important for avoiding graph runs with potentially infinite loops and thus minimizing unnecessary costs."